src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -1,36 +1,128 @@
+ * Uses ratchet mechanism to break
+ * 
+ * What Modes Does It Have? 1) Up (Extend) 2) Down (Retract) 3) Brake (Ratchet)
+ * 4) Hook onto Bar using d-clip 5) Go to Middle Bar Position 6) Go to Top Bar
+ * Position
+ * 
+ * What Interactions Does It Have With Other Subsystems? 1) None
+ * 
+ * How Is It Triggered? 1) User Input
+ * 
+ * Does It Store Any State? 1) isBraked 2) Limit switch values 3) Encoder values
+ * 
+ * Sensors? 1) Encoders 2) Limit Switches
+ */
 
-    private SendableSpeedController elevatorMotor;
+public class Lift extends SubsystemBase {
 
+    private PIDSpeedController elevatorMotor;
+    private ISolenoid elevatorBrake;
+    private IEncoder liftEncoder;
+    private BooleanSupplier upperLimitSwitch;
+    private BooleanSupplier lowerLimitSwitch;
     private static final double elevatorMotorSpeed = 1;
+    private static final double TOLERANCE_RANGE_INCHES = .5;
+    // TODO I don't know what unit tolerance range is in but I set it to .5 assuming
+    // it's in inches
 
+    // TODO Find a way to sync the elevatorLeft and elevatorRight motors
     public Lift(RobotMap.LiftMap map) {
         super();
         elevatorMotor = map.elevator();
+        liftEncoder = map.getLiftEncoder();
+        elevatorBrake = map.liftBrake();
+        upperLimitSwitch = map.upperLiftLimit();
+    }
+
+    // sets the ratchet to either be activated or deactivated depending on liftSpeed
+    public void liftSpeed(double speed) {
+        if (speed > 0) {
+            elevatorBrake.set(false);
+            if (upperLimitSwitch.getAsBoolean()) {
+                speed = 0;
+            }
+        } else if (speed <= 0) {
+            elevatorBrake.set(true);
+            if (lowerLimitSwitch.getAsBoolean()) {
+                speed = 0;
+            }
+        }
+        elevatorMotor.set(speed);
+    }
+
+    // Heights of the bar at max, min, and middle
+    public enum liftHeights {
+        Top(52.25), Middle(63), Bottom(78.125);
+
+        private double mPosition;
+
+        private double minLiftHeight = 51;
+
+        // Returning an double to compare whether we're in the right place or not
+        public double value() {
+            return this.mPosition - minLiftHeight;
+        }
+
+        // Returning the level the lift is at (top middle or bottom)
+        private liftHeights(double mPosition) {
+            this.mPosition = mPosition;
+        }
     }
 
-    public CommandBase up() {
-        return new StartEndCommand(() -> {
-            elevatorMotor.set(-elevatorMotorSpeed);
+    public CommandBase moveLift(DoubleSupplier speed) {
+        // The command is named "Move Lift" and requires this subsystem.
+        return new FunctionalCommand(() -> {
+            liftSpeed(0);
+        }, () -> {
+            liftSpeed(speed.getAsDouble());
+        }, (Boolean interrupted) -> {
+            liftSpeed(0);
         }, () -> {
-            elevatorMotor.stopMotor();
+            return false;
```

Comment:
Fun fact: `() -> { return false; }` can be shortened to `() -> false`. Nice little (optional) shorthand.